replit_river/session.py

```diff
@@ -382,9 +384,11 @@ async def send_message(
             serviceName=service_name,
             procedureName=procedure_name,
         )
-        TraceContextTextMapPropagator().inject(
-            msg, None, TransportMessageTracingSetter()
-        )
+        if span:
+            with use_span(span):
+                TraceContextTextMapPropagator().inject(
```

Comment:
would it be worth it (or is it advantageous) to cache these propagators/setters?

Reply:
Yeah we can probably just save these as globals, they don't store any state or anything, just implement an interface.